GUIDE.md

```diff
@@ -1,28 +1,217 @@
-# Server Endpoint
+# Guide
 
+ - The guiding focus to how to write a client to connect to a Tokyo server, receive information events and create actions to control your ship
+
+## 1. Connection information
+
+`ws://${host}/socket?key={key}&name{name}`
+
+| Parameters | Description |
+|--|--|
+| {host} | Domain or IP of server want to connect |
+| {key} | User's identity is unique and used to distinguish bots |
+| {name} | The display name will be shown on UI |
+
+When the WebSocket connection is established successfully, your bot is registered, displayed on web UI and ready to use.
+
+## 2. Events
+
+From WebSocket, the server consecutively sends events to the client every tick with the following structure.
+Event message structure contains all world info of the game, including players, bullets, dead person, etc.
+
+### 2.1. Events structure
+
+#### 2.1.1. State event
+
+State event contains world map states that includes map info, players, bullets, dead and scoreboard
+
+```json
+{
+  "e": "state",
+  "data": {
+    "bounds": [1200.0, 800.0],
+    "players": [
+      {},
+      {}
+    ],
+    "bullets": [
+      {},
+      {}
+    ],
+    "dead": [
+      "respawn":{
+         "secs_since_epoch":1631299999,
+         "nanos_since_epoch":940876053
+       },
+       "player": {}
+    ],
+    "scoreboard":{"0":100,"1":90,"2":80}
+  }
+}
+```
+
+| Fields | Description |
+|--|--|
+| e | State event is is always "state" |
+| data | Detail data of event "e" |
+| bounds | Boundary of the game, players spawn and navigate their ship in boundary from position [0,0] to this max size boundary. It's an array with two values, width and height |
+| players | List of players/ships in the game currently. Detail of the player object will be described in the next sections |
+| bullets | List of bullets that's fired by ships in the game currently. Detail of bullet object will be described in the next sections |
+| dead | List of dead users and the respawn periods. Information of player is a structure with "players" |
+| scoreboard | Top user scores with format "player_id: score" |
+
+#### 2.1.2. Current user event
+
+Event contains current user id
+
+```json
+{"e":"id","data":247}
+```
+
+| Fields | Description |
+|--|--|
+| e | State event is is always "state" |
+| data | id of current user |
+
+#### 2.1.3. User event
+
+Event contains all users and their ids in the game.
+
+```json
+{
+   "e":"teamnames",
+   "data":{
+      "60":"thich_khanh_ngoc",
+      "178":"z",
+      "206":"chicken_killer",
+      "222":"tuan",
+      "161":"don't kill me",
+      "240":"coward_dog",
+      "244":"do_anh_bat_duoc_em",
+      "134":"thich_khanh_ngoc",
+      "154":"tuan",
+      "175":"hieuk09",
+      "176":"z",
+      "173":"don't kill me",
+      "229":"tuan",
+   }
+}
 ```
-$HOST/socket
+
+| Fields | Description |
+|--|--|
+| e | State event is is always "state" |
```

Comment:
Teamnames event is always `teamnames` instead of `state`

Reply:
my wrong